src/libAtomVM/opcodesswitch.h

```diff
@@ -5836,6 +5828,349 @@ static bool maybe_call_native(Context *ctx, AtomString module_name, AtomString f
+                for (int j = 0;  j < nb_segments; j++) {
+                    term atom_type;
+                    DECODE_ATOM(atom_type, code, i, next_off);
+                    int seg;
+                    DECODE_LITERAL(seg, code, i, next_off);
+                    int segment_unit;
+                    DECODE_LITERAL(segment_unit, code, i, next_off);
+                    term flags;
+                    DECODE_COMPACT_TERM(flags, code, i, next_off);
+                    term src;
+                    DECODE_COMPACT_TERM(src, code, i, next_off);
+                    term size;
+                    DECODE_COMPACT_TERM(size, code, i, next_off);
+                    #ifdef IMPL_EXECUTE_LOOP
+                        avm_int_t src_value = 0;
+                        switch (atom_type) {
+                            case UTF8_ATOM:
+                            case UTF16_ATOM:
+                            case UTF32_ATOM:
+                                VERIFY_IS_INTEGER(src, "bs_create_bin/6");
+                                src_value = term_to_int(src);
+                                break;
+                        }
+                        size_t segment_size = 0;
+                        switch (size) {
+                            case UNDEFINED_ATOM: {
+                                // Silently ignore segment_unit != 0
+                                segment_unit = 8;
+                                switch (atom_type) {
+                                    case UTF8_ATOM: {
+                                        if (UNLIKELY(!bitstring_utf8_size(src_value, &segment_size))) {
+                                            RAISE_ERROR(BADARG_ATOM);
+                                        }
+                                        break;
+                                    }
+                                    case UTF16_ATOM: {
+                                        if (UNLIKELY(!bitstring_utf16_size(src_value, &segment_size))) {
+                                            RAISE_ERROR(BADARG_ATOM);
+                                        }
+                                        break;
+                                    }
+                                    case UTF32_ATOM: {
+                                        segment_size = 4;
+                                        break;
+                                    }
+                                    default:
+                                        // In Erlang/OTP #5281, this is a compile time check
+                                        fprintf(stderr, "Unexpected type %lx for bs_create_bin/6 size undefined\n", (long) atom_type);
+                                        AVM_ABORT();
+                                }
+                                break;
+                            }
+                            case ALL_ATOM: {
+                                if (atom_type != BINARY_ATOM && atom_type != APPEND_ATOM && atom_type != PRIVATE_APPEND_ATOM) {
```

Comment:
I guess my ask is to instantiate the `binary` and `append` atom (terms) here, instead of in defaultplatforms.c, since doing instantiating in default platforms causes the atom to occupy space in the atoms table, even if this instruction is not executed in the program.

I wonder if that means we should `extern` the AtomStrings that are instantiated in defaultatoms.c, so that the `binary_atom`, `append_atom`, etc, are accessible so that they don’t have to be re-instantiated in the code/data section of the image.

Does that make sense?

Reply:
All atoms in `opcodesswitch.h` should be in defaultatoms for the following reasons:
- speed: they are constants
- simplicity: no need to worry about malloc failures
- they are still in a limited an reasonable number overall